src/model/encoding/convertFromHTMLToContentBlocks.js

```diff
@@ -116,23 +118,76 @@ function getBlockDividerChunk(block: DraftBlockType, depth: number): Chunk {
   };
 }
 
-function getBlockTypeForTag(tag: string, lastList: ?string): DraftBlockType {
-  switch (tag) {
-    case 'h1':
-      return 'header-one';
-    case 'h2':
-      return 'header-two';
-    case 'li':
+function getListBlockType(
+  tag: string,
+  lastList: ?string
+): ?DraftBlockType {
+  if(tag === 'li') {
       if (lastList === 'ol') {
         return 'ordered-list-item';
       }
       return 'unordered-list-item';
-    case 'blockquote':
-      return 'blockquote';
-    case 'pre':
-      return 'code-block';
+  }
+
+  return null;
+}
+
+function getBlockMapSupportedTags(
+  draftBlockMap: DraftBlockMap
+): Array<DraftBlockType> {
+  const blockTypes = Object.keys(draftBlockMap);
+  const matchedTags = blockTypes.map(function(blockType) {
+      return draftBlockMap[blockType].element;
+  }).sort();
+
+  const supportedTags = matchedTags.filter(function(tag, index) {
+      const notUnstyledTag = tag !== draftBlockMap.unstyled.element;  // we do not want unstyled to be viewed as a valid match
+      const uniqueTag = matchedTags.indexOf(tag) === index;  // this will guarantee no duplicate tags added
+      return notUnstyledTag && uniqueTag;
+  });
+
+  return supportedTags;
+}
+
+// custom element conversions
+function getMultiMatchedType(
+  tag: string,
+  lastList: ?string,
+  multiMatchExtractor: Array<Function>
+): ?DraftBlockType {
+    for (var i = 0; i < multiMatchExtractor.length; i++) {
+        const matchType = multiMatchExtractor[i](tag, lastList);
+
+        if (matchType) {
+            return matchType;
+        }
+    }
+
+    return null;
+}
+
+function getBlockTypeForTag(
+  tag: string,
+  lastList: ?string,
+  draftBlockMap: DraftBlockMap
+): DraftBlockType {
+  const blockTypes = Object.keys(draftBlockMap);
+  const matchedTypes = blockTypes.filter(function(blockType) {
+      return draftBlockMap[blockType].element === tag || draftBlockMap[blockType].wrapper === tag;
+  });
+  const matchesFound = matchedTypes.length;
+  const UNSTYLED = 'unstyled';
+
+  // if we dont have any matched type, return unstyled
+  // if we have one matched type return it
+  // if we have multi matched types use the multi-match function to determine which type is it
+  switch(matchesFound) {
+    case 0:
+        return UNSTYLED;
+    case 1:
+        return matchedTypes[0];
     default:
-      return 'unstyled';
+        return getMultiMatchedType(tag, lastList, [getListBlockType]) || UNSTYLED;
```

Comment:
One of the assumptions of allowing users to define a Custom block map, is that they have to always have a 'unstyled' block map.